packages/big-design/src/components/Tree/TreeNode/TreeNode.tsx

```diff
@@ -21,268 +21,134 @@ import {
-  treeId,
   value,
-}: TreeNodeProps<T> & PrivateTreeItemProps<T>): React.ReactElement<TreeNodeProps<T>> => {
-  const thisRef = useRef<TreeNodeRef<T>>({ children });
+  id,
+}: TreeNodeProps<T>): React.ReactElement<TreeNodeProps<T>> => {
+  const { disabledNodes, expandable, focusable, iconless, onKeyDown, onNodeClick, selectable } = useContext(
+    TreeContext,
+  );
   const nodeRef = useRef<HTMLLIElement | null>(null);
   const selectableRef = useRef<HTMLLabelElement | null>(null);
-  const treeNodeId = `${treeId}-treenode-${id}`;
   const [isLoading, setIsLoading] = useState(false);
-  const expanded = useIsExpanded(state, id);
-  const selected = useIsSelected(state, value);
-  const selectedChildrenCount = useSelectedChildrenCount(state, children);
+  const isExpanded = expandable.expandedNodes.includes(id);
+  const isSelected = selectable?.selectedNodes?.includes(id);
+  const isDisabled = disabledNodes?.includes(id);
+  const isSelectable = value !== undefined && selectable?.type !== undefined && !isDisabled;
+  const selectedChildrenCount = useSelectedChildrenCount({ selectedNodes: selectable?.selectedNodes, children });
 
-  // Could be multiple elements in which are clicked.
-  // Typing to generic Element type since all other elements extend from it.
-  const handleNodeToggle = async (e?: React.MouseEvent<Element>) => {
-    dispatch({ type: 'FOCUS', id });
-
-    // Prevents the collapse/expand when clicking on a radio or checkbox
-    // Checks to see if every element inside the selectableRef gets clicked.
+  useEffect(() => {
     if (
-      (e?.target instanceof Node && selectableRef.current?.contains(e?.target)) ||
-      thisRef.current.children === undefined
+      focusable.focusedNode === id &&
+      nodeRef.current !== document.activeElement &&
+      document.activeElement !== document.body
     ) {
-      return;
+      nodeRef.current?.focus();
     }
+  }, [focusable, id]);
 
-    dispatch({ type: 'TOGGLE_NODE', id });
-
-    if (expanded) {
-      if (typeof onCollapse === 'function') {
-        const callbackValue = await onCollapse({
-          children: thisRef.current.children,
-          disabled,
-          expanded,
-          id,
-          label,
-          selected,
-          value,
-        });
+  // Could be multiple elements in which are clicked.
+  // Typing to generic Element type since all other elements extend from it.
+  const handleNodeToggle = useCallback(
+    async (e?: React.MouseEvent<Element>) => {
+      // Prevents the collapse/expand when clicking on a radio or checkbox
+      // Checks to see if every element inside the selectableRef gets clicked.
+      if ((e?.target instanceof Node && selectableRef.current?.contains(e?.target)) || children === undefined) {
+        return;
+      }
 
-        if (callbackValue) {
-          thisRef.current = callbackValue;
-        }
+      if (typeof expandable.onToggle === 'function') {
+        await expandable.onToggle(id, isExpanded);
```

Comment:
`await` is not doing much here, unless the types are off for `onToggle`, `onCollapse`, and `onExpand`.

Reply:
Remnant of the old method of lazy-loading children. I'll most likely rip that out for now.